lib/ansible/plugins/inventory/ovirt.py

```diff
@@ -0,0 +1,275 @@
+        type: list
+'''
+
+EXAMPLES = '''
+# Sample content of ovirt.yml
+# plugin: ovirt
+# ovirt_url: http://localhost/ovirt-engine/api
+# ovirt_username: ansible-tester
+# ovirt_password: secure
+# ovirt_query_filter:
+#   search: 'name=myvm AND cluster=mycluster'
+#   case_sensitive: no
+#   max: 15
+# keyed_groups:
+#   - key: cluster
+#     prefix: 'cluster'
+# groups:
+#   dev: "'dev' in tags"
+# compose:
+#   ansible_host: devices["eth0"][0]
+'''
+
+import sys
+
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+from ansible.errors import AnsibleError, AnsibleParserError
+
+HAS_OVIRT_LIB = False
+
+try:
+    import ovirtsdk4 as sdk
+    HAS_OVIRT_LIB = True
+except ImportError:
+    HAS_OVIRT_LIB = False
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'ovirt'
+
+    def __init__(self):
+
+        super(InventoryModule, self).__init__()
+        self.connection = None
+
+    def _get_connection(self):
+        if self.connection is None:
+            self.connection = sdk.Connection(
+                url=self.ovirt_engine_url,
+                username=self.ovirt_engine_user,
+                password=self.ovirt_engine_password,
+                ca_file=self.ovirt_engine_cafile
+            )
+        return self.connection
+
+    def _get_dict_of_struct(self, vm):
+        '''  Transform SDK Vm Struct type to Python dictionary.
+             :param vm: host struct of which to create dict
+             :return dict of vm struct type
+        '''
+
+        connection = self._get_connection()
+
+        vms_service = connection.system_service().vms_service()
+        clusters_service = connection.system_service().clusters_service()
+        vm_service = vms_service.vm_service(vm.id)
+        devices = vm_service.reported_devices_service().list()
+        tags = vm_service.tags_service().list()
+        stats = vm_service.statistics_service().list()
+        labels = vm_service.affinity_labels_service().list()
+        groups = clusters_service.cluster_service(
+            vm.cluster.id
+        ).affinity_groups_service().list()
+
+        return {
+            'id': vm.id,
+            'name': vm.name,
+            'host': connection.follow_link(vm.host).name if vm.host else None,
+            'cluster': connection.follow_link(vm.cluster).name,
+            'status': str(vm.status),
+            'description': vm.description,
+            'fqdn': vm.fqdn,
+            'os_type': vm.os.type,
+            'template': connection.follow_link(vm.template).name,
+            'tags': [tag.name for tag in tags],
+            'affinity_labels': [label.name for label in labels],
+            'affinity_groups': [
+                group.name for group in groups
+                if vm.name in [vm.name for vm in connection.follow_link(group.vms)]
+            ],
+            'statistics': dict(
+                (stat.name, stat.values[0].datum) for stat in stats
```

Comment:
Had some issues with this in my env but this fixed it:
`(stat.name, stat.values[0].datum if stat.values else None) for stat in stats`